posit-dev/positron
# Connections pane integration with the variables pane

### Intent

This PR integrates the variables pane with the connections pane. When variables are assigned and they are known to the connections pane we display a small icon, that triggers the registration of that connection with the connections pane.
Addresses https://github.com/posit-dev/positron/issues/2503


### Approach

> Describe the approach taken and the tradeoffs involved if non-obvious; add an overview of the solution if it's complicated.

1. Added a new inspector that catches supported connections and enables the `view` buttom for them:

https://github.com/posit-dev/positron/blob/f233f3ee41366f6f9110abc16310e3418f4b7300/extensions/positron-python/pythonFiles/positron/positron_ipykernel/inspectors.py#L902-L909

2. Made the variables pane preview handling depend on the type of object, and for supported values we will
    enable register the object with the connections service:

https://github.com/posit-dev/positron/blob/f233f3ee41366f6f9110abc16310e3418f4b7300/extensions/positron-python/pythonFiles/positron/positron_ipykernel/variables.py#L519-L537

 3. Add listeners to variable updates/deletions in the variables service allowing the connections service to handle those events, if required.

https://github.com/posit-dev/positron/blob/f233f3ee41366f6f9110abc16310e3418f4b7300/extensions/positron-python/pythonFiles/positron/positron_ipykernel/variables.py#L157-L169


### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues.

A small video below showing different scenarios that we expect this PR to work with:

https://github.com/posit-dev/positron/assets/4706822/a08050c9-b84d-4b35-a804-fe051f201b9d
